26_Autoexposure/main.cpp

```diff
@@ -0,0 +1,874 @@
+// Copyright (C) 2018-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#include "nbl/application_templates/MonoAssetManagerAndBuiltinResourceApplication.hpp"
+#include "../common/SimpleWindowedApplication.hpp"
+
+#include "nbl/video/surface/CSurfaceVulkan.h"
+#include "nbl/asset/interchange/IAssetLoader.h"
+#include "nbl/ext/FullScreenTriangle/FullScreenTriangle.h"
+
+#include "app_resources/common.hlsl"
+
+using namespace nbl;
+using namespace core;
+using namespace hlsl;
+using namespace system;
+using namespace asset;
+using namespace ui;
+using namespace video;
+
+class AutoexposureApp final : public examples::SimpleWindowedApplication, public application_templates::MonoAssetManagerAndBuiltinResourceApplication
+{
+	using device_base_t = examples::SimpleWindowedApplication;
+	using asset_base_t = application_templates::MonoAssetManagerAndBuiltinResourceApplication;
+	using clock_t = std::chrono::steady_clock;
+
+	constexpr static inline std::string_view DefaultImagePathsFile = "../../media/noises/spp_benchmark_4k_512.exr";
+	constexpr static inline std::array<uint32_t, 2> Dimensions = { 1280, 720 };
+	constexpr static inline std::array<uint32_t, 2> SampleCount = { 10000, 10000 };
```

Comment:
sample count is useless, you should
1. deduce the workgroup count needed in the dispatch from `Dimensions*MeteringWindowScale`
2. then deduce your TOTAL sample count from the workgroup counts